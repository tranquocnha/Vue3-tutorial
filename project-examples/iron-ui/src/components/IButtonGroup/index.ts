export { default as IButtonGroup } from '@inkline/inkline/components/IButtonGroup/IButtonGroup.vue';
