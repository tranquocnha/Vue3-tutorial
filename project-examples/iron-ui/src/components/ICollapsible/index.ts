export { default as ICollapsible } from '@inkline/inkline/components/ICollapsible/ICollapsible.vue';
