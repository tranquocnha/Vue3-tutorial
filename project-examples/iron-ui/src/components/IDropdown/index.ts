export { default as IDropdown } from '@inkline/inkline/components/IDropdown/IDropdown.vue';
