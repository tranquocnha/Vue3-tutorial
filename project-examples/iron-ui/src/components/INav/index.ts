export { default as INav } from '@inkline/inkline/components/INav/INav.vue';
