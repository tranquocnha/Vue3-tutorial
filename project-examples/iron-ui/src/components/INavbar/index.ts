export { default as INavbar } from '@inkline/inkline/components/INavbar/INavbar.vue';
