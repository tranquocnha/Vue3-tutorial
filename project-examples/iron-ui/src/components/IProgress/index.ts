export { default as IProgress } from '@inkline/inkline/components/IProgress/IProgress.vue';
