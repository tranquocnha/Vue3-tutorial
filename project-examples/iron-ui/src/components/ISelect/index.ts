export { default as ISelect } from '@inkline/inkline/components/ISelect/ISelect.vue';
