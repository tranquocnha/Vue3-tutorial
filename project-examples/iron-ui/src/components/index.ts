// Composition API

export { IAlert } from '@inkline/inkline/components/IAlert';
export { IBadge } from '@inkline/inkline/components/IBadge';
export { IBreadcrumb } from '@inkline/inkline/components/IBreadcrumb';
export { IBreadcrumbItem } from '@inkline/inkline/components/IBreadcrumbItem';
export { IButton } from '@inkline/inkline/components/IButton';
export { IButtonGroup } from '@inkline/inkline/components/IButtonGroup';
export { ICard } from '@inkline/inkline/components/ICard';
export { ICheckbox } from '@inkline/inkline/components/ICheckbox';
export { ICheckboxGroup } from '@inkline/inkline/components/ICheckboxGroup';
export { ICollapsible } from '@inkline/inkline/components/ICollapsible';
export { ICollapsibleItem } from '@inkline/inkline/components/ICollapsibleItem';
export { IColumn } from '@inkline/inkline/components/IColumn';
export { IContainer } from '@inkline/inkline/components/IContainer';
export { IDropdown } from '@inkline/inkline/components/IDropdown';
export { IDropdownDivider } from '@inkline/inkline/components/IDropdownDivider';
export { IDropdownItem } from '@inkline/inkline/components/IDropdownItem';
export { IExpandTransition } from '@inkline/inkline/components/IExpandTransition';
export { IForm } from '@inkline/inkline/components/IForm';
export { IFormGroup } from '@inkline/inkline/components/IFormGroup';
export { IFormError } from '@inkline/inkline/components/IFormError';
export { IFormLabel } from '@inkline/inkline/components/IFormLabel';
export { IHamburgerMenu } from '@inkline/inkline/components/IHamburgerMenu';
export { IHeader } from '@inkline/inkline/components/IHeader';
export { IIcon } from '@inkline/inkline/components/IIcon';
export { IInput } from '@inkline/inkline/components/IInput';
export { ILayout } from '@inkline/inkline/components/ILayout';
export { ILayoutAside } from '@inkline/inkline/components/ILayoutAside';
export { ILayoutContent } from '@inkline/inkline/components/ILayoutContent';
export { ILayoutFooter } from '@inkline/inkline/components/ILayoutFooter';
export { ILayoutHeader } from '@inkline/inkline/components/ILayoutHeader';
export { IListGroup } from '@inkline/inkline/components/IListGroup';
export { IListGroupItem } from '@inkline/inkline/components/IListGroupItem';
export { ILoader } from '@inkline/inkline/components/ILoader';
export { IMark } from '@inkline/inkline/components/IMark';
export { IMedia } from '@inkline/inkline/components/IMedia';
export { IModal } from '@inkline/inkline/components/IModal';
export { INav } from '@inkline/inkline/components/INav';
export { INavItem } from '@inkline/inkline/components/INavItem';
export { INavbar } from '@inkline/inkline/components/INavbar';
export { INavbarBrand } from '@inkline/inkline/components/INavbarBrand';
export { INavbarCollapsible } from '@inkline/inkline/components/INavbarCollapsible';
export { INumberInput } from '@inkline/inkline/components/INumberInput';
export { IPagination } from '@inkline/inkline/components/IPagination';
export { IPopover } from '@inkline/inkline/components/IPopover';
export { IProgress } from '@inkline/inkline/components/IProgress';
export { IProgressBar } from '@inkline/inkline/components/IProgressBar';
export { IRadio } from '@inkline/inkline/components/IRadio';
export { IRadioGroup } from '@inkline/inkline/components/IRadioGroup';
export { IRow } from '@inkline/inkline/components/IRow';
export { ISelect } from '@inkline/inkline/components/ISelect';
export { ISelectOption } from '@inkline/inkline/components/ISelectOption';
export { ISidebar } from '@inkline/inkline/components/ISidebar';
export { ITable } from '@inkline/inkline/components/ITable';
export { ITabs } from '@inkline/inkline/components/ITabs';
export { ITab } from '@inkline/inkline/components/ITab';
export { ITabTitle } from '@inkline/inkline/components/ITabTitle';
export { ITextarea } from '@inkline/inkline/components/ITextarea';
export { IToast } from '@inkline/inkline/components/IToast';
export { IToastContainer } from '@inkline/inkline/components/IToastContainer';
export { IToggle } from '@inkline/inkline/components/IToggle';
export { ITooltip } from '@inkline/inkline/components/ITooltip';
