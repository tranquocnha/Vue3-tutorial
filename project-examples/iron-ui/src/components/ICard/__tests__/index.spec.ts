import { render } from '@testing-library/vue';
import { ICard } from '@inkline/inkline/components';
import { InklineKey } from '@inkline/inkline/plugin';
import { createInkline } from '@inkline/inkline/__tests__/utils';

describe('Components', () => {
    describe('ICard', () => {
        const props = {
            color: 'light',
            size: 'md'
        };

        it('should be named correctly', () => {
            expect(ICard.name).toEqual('ICard');
        });

        it('should render correctly', () => {
            const wrapper = render(ICard, {
                props,
                global: {
                    provide: {
                        [InklineKey as symbol]: createInkline()
                    }
                }
            });
            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('computed', () => {
            describe('classes', () => {
                it('should add classes based on props', () => {
                    const wrapper = render(ICard, {
                        props,
                        global: {
                            provide: {
                                [InklineKey as symbol]: createInkline()
                            }
                        }
                    });

                    expect(wrapper.container.firstChild).toHaveClass(
                        `-${props.color}`,
                        `-${props.size}`
                    );
                });
            });
        });
    });
});
