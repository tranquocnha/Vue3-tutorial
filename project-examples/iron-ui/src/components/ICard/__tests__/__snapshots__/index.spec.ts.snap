// Vitest Snapshot v1

exports[`Components > ICard > should render correctly 1`] = `
"<div class=\\"card -light -md\\">
  <!--v-if-->
  <!-- @slot image Slot for card image -->
  <!--v-if-->
  <!--v-if-->
</div>"
`;
